### PR TITLE
feat: add YNAP parsing to transaction import

### PR DESCRIPTION
**Pull request by morremeyer**
This adds YNAP parsing to transaction importing, removing the need to
parse the file in the YNAP web app before uploading it to Envelope Zero.

Obsoletes #1043.
Part of #936.

### CODE REVIEW

File: cypress/e2e/transaction-import.cy.ts [hunk 1]
```diff
@@ -163,11 +161,43 @@ describe('Transaction Import', () => {
...
+
+    cy.getInputFor('Account').type('account')
+    cy.contains('External Account').should('not.exist')
+    cy.contains('My Other Account')
+    cy.contains('My Account').click()
+
+    cy.getInputFor('File').selectFile('cypress/fixtures/comdirect-ynap.csv')
+
+    cy.contains('Submit').click()
+    cy.contains('Error: This file has already been converted')
```
Comment (malfynnction) [inline cypress/e2e/transaction-import.cy.ts:183]: We should probably display a more user-friendly error message. This doesn't really explain what went wrong and how to fix it.

File: cypress/e2e/transaction-import.cy.ts [hunk 2]
```diff
@@ -163,11 +161,43 @@ describe('Transaction Import', () => {
...
+    cy.contains('External Account').should('not.exist')
+    cy.contains('My Other Account')
+    cy.contains('My Account').click()
+
+    cy.getInputFor('File').selectFile(
+      'cypress/fixtures/transactions-no-parser.csv'
+    )
+
+    cy.contains('Submit').click()
+    cy.contains('Error: No parser is available for this file')
```
Comment (malfynnction) [inline cypress/e2e/transaction-import.cy.ts:201]: Same here. It should say something like "we can't parse this file, please check here (link to ynap) if your bank is supported"

File: cypress/fixtures/transactions-no-parser.csv
Comment (malfynnction) [inline cypress/fixtures/transactions-no-parser.csv:1]: What's the difference between this file and `cypress/fixtures/comdirect-ynap.csv`?

File: src/components/TransactionImport/Form.tsx [hunk 3]
```diff
@@ -29,23 +30,53 @@ const Form = ({ accounts, isLoading, setIsLoading, setResult }: Props) => {
...
-              setError('')
-            }
-            setResult(body.data, accountId)
+        parseFile(file)
+          .then(result => {
+            // Get the data and transform it to FormData for the backend
+            const data = new FormData()
+            data.append(
+              'file',
+              new File([new Blob([result[0].data])], 'ynap-transactions.csv', {
```
Comment (malfynnction) [inline src/components/TransactionImport/Form.tsx:41]: To ensure we don't crash if `result` doesn't look as expected:

```suggestion
              new File([new Blob([result[0]?.data])], 'ynap-transactions.csv', {
```

File: src/translations/en.json [hunk 4]
```diff
@@ -88,20 +88,15 @@
...
       "duplicateDetected": "This is a duplicate of an existing transaction",
       "complete": "Import complete",
-      "howTo": {
-        "title": "How To Import",
-        "download": "Download a .CSV export of your transactions from your bank",
-        "parse": "Parse that file using <a class='link-blue' href='https://ynap.leolabs.org/'>YNAP</a> (A list of all currently supported banks can be found <a class='link-blue' href='https://ynap.leolabs.org/supported-formats/'>here</a>)",
-        "upload": "Upload the parsed file here:"
-      }
+      "alreadyConverted": "This file has already been parsed. Please upload the CSV file from your bank.",
+      "noParser": "This file can not be parsed. Please check on https://l.envelope-zero.org/ynap-formats if your bank is supported."
```
Comment (malfynnction) [inline src/translations/en.json:99]: Let's avoid non-clickable links
```suggestion
      "noParser": "This file can not be parsed. Please check the link below to see if your bank is supported."
```

File: cypress/fixtures/transactions-no-parser.csv
Comment (morremeyer) [reply cypress/fixtures/transactions-no-parser.csv:1]: *ynap-parsers* detects already parsed files by matching the file name against `*-ynap.csv`. The content of the files is the same, they're just used to detect different error scenarios.

File: src/translations/en.json (hunk 4 above)
Comment (morremeyer) [reply src/translations/en.json]: See 8221cd1f25f4d49d8b0a1c199806c04e8d8b1cb2

**Review by malfynnction — APPROVED** [review]
🚀

**Pull request merged**